curated_transformers/repository/repository.py

```diff
@@ -0,0 +1,279 @@
+                ) from e
+
+        return self._model_config
+
+    def model_type(self) -> str:
+        """
+        Get the model type.
+
+        :returns:
+            The model type.
+        :raises OSError:
+            When the model config cannot be opened.
+        """
+        return self.model_config()["model_type"]
+
+    def pretty_path(self, path: Optional[str] = None) -> str:
+        return self.repo.pretty_path(path)
+
+
+class TokenizerRepository(Repository):
+    """
+    Repository wrapper that exposes some methods that are useful for working
+    with repositories that contain a tokenizer.
+    """
+
+    _tokenizer_config: Optional[Dict[str, Any]]
+
+    def __init__(self, repo: Repository):
+        self.repo = repo
+        self._tokenizer_config = None
+
+    def file(self, path: str) -> RepositoryFile:
+        return self.repo.file(path)
+
+    def json_file(self, path: str) -> Dict[str, Any]:
+        return self.repo.json_file(path)
+
+    def model_type(self) -> str:
+        """
+        Get the model type.
+
+        :returns:
+            The model type.
+        """
+        model_config = self.json_file(HF_MODEL_CONFIG)
+        return model_config["model_type"]
+
+    def pretty_path(self, path: Optional[str] = None) -> str:
+        return self.repo.pretty_path(path)
+
+    def special_tokens_map(self) -> Dict[str, Any]:
+        """
+        Return the tokenizer's special tokens map.
+
+        :returns:
+            The special tokens map.
+        :raises OSError:
+            When the special tokens map cannot be opened.
+        """
+        try:
+            return self.repo.json_file(
+                path=SPECIAL_TOKENS_MAP,
+            )
+        except (FileNotFoundError, JSONDecodeError, OSError) as e:
+            raise OSError(
+                "Could not load special tokens map for the tokenizer at "
+                f"`{self.repo.pretty_path(SPECIAL_TOKENS_MAP)}`"
+            ) from e
+
+    def tokenizer_config(self) -> Dict[str, Any]:
+        """
+        Return the model's tokenizer configuration. The result is cached to
+        speed up subsequent lookups.
+
+        :returns:
+            Model configuration.
+        :raises OSError:
+            When the tokenizer config cannot be opened.
+        """
+        if self._tokenizer_config is None:
+            try:
+                self._tokenizer_config = self.repo.json_file(
+                    path=HF_TOKENIZER_CONFIG,
+                )
+            except (FileNotFoundError, JSONDecodeError, OSError) as e:
+                raise OSError(
+                    "Couldn't find a valid config for the tokenizer at "
+                    f"`{self.repo.pretty_path(HF_TOKENIZER_CONFIG)}`"
+                ) from e
+
+        return self._tokenizer_config
+
+    def tokenizer_json(self) -> RepositoryFile:
+        """
+        Return the HF tokenizers' ``tokenizer.json``.
+
+        :returns:
+            The tokenizer file.
```

Comment:
```suggestion
            The repository file.
```

Reply:
But the initial description is more accurate, right? We are not just returning any repository file, we are returning a file that is the actual tokenizer(.json).

Reply:
On second thought, yeah, it's less descriptive than the original wording. Never mind.